### ISSUE TITLE
Document SQLite's lack of BigInt primary key support

### ISSUE THREAD

**Issue by mahendraHegde** [directus/docs#311]
### Describe the Bug
Choosing autoincrement bigInt as id actually creates integer auto increment.

### To Reproduce

1. select sqlite driver
2. create a new collection with id as bigInt with auto increment
3. go back to field definition in UI and see, it will be reset to integer

### Directus Version

10.13.3

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by br41nslug** [comment]
I dont think this has anything to do with SQLite specifically but rather just BigInt handling in filters in general.

I am unfortunately not able to reproduce the described resetting of the stored type in SQLite. Are there extra steps required for this reset to happen? I've tried navigating and restarting the instance

https://github.com/directus/docs/assets/9389634/bfbe5924-20aa-46f4-9604-b9d1202c8e37

Closing this for now until we can get this reproduceable.

P.S.
I thought there was already an issue for this on the API level but cannot find it, perhaps we'll need to retrofit this issue to be generic to the API instead.

**Comment by mahendraHegde** [comment]
https://screenapp.io/app/#/shared/d98c829a-7383-4b0e-ae44-c84f4a021641 
@br41nslug  it seems it only happening for bigInt autoincrement
initially i thought its an issue with knex, as query generated was ```create table `test2` (`id` integer not null primary key autoincrement)'```

but when i tried directly ```create table `test2` (`id` bigint not null primary key autoincrement)'```  in sql console sqlite rejects it with error `AUTOINCREMENT is only allowed on an INTEGER PRIMARY KEY`

**Comment by br41nslug** [comment]
You're very right explicitly when using it as primary key i was able to reproduce it 😄 Sounds like SQLite simply does not support it tho 🤔 SQLite does not support this because they make no distinction between `integer` and `bigint` like other vendors, in sqlite's case they're both int64. https://sqlite.org/forum/forumpost/415111b8e26732ec5f537912a9c6e63394a86bad0e7e831055e3234843b61e02

I am not exactly sure what the correct "solution" is here:
- falling back to `integer` gracefully (which is what we do now, and recommended by sqlite)
- throwing an error that you cannot select "big int autoincrement" keys for sqlite
- just documenting this edgecase for sqlite

**Comment by mahendraHegde** [comment]
also may be 
-  disabling/hiding bigInt increment option in ui for sqlite

**Comment by br41nslug** [comment]
> disabling/hiding bigInt increment option in ui for sqlite

I don't think we should go that route, the App itself is not aware of what backend database was used and shouldn't be i think

**Comment by phazonoverload** [comment]
The right place for this note in docs will be the yet-to-be created directus/docs#310 Should we consider this a docs issue @br41nslug?

**Comment by br41nslug** [comment]
Yeah i think educating about this is the best way to go here since this is such an edge case and it is still working as intended because SQLite makes no distinction between `int` or `bigInt`.
